spec/features/internal_spec.rb

```diff
@@ -0,0 +1,179 @@
+require "rails_helper"
+require "action_text/system_test_helper"
+
+RSpec.feature "Internal section", type: :feature do
+  let(:types) { Events::Search.available_event_type_ids }
+  let(:event) do
+    build(:event_api,
+          :with_provider_info,
+          name: "Pending event",
+          status_id: GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"])
+  end
+  let(:events_by_type) { group_events_by_type([event]) }
+
+  before do
+    if page.driver.browser.respond_to?(:authorize)
+      page.driver.browser.authorize(ENV["PUBLISHER_USERNAME"], ENV["PUBLISHER_PASSWORD"])
+    end
+
+    allow_any_instance_of(GetIntoTeachingApiClient::TeachingEventsApi).to \
+      receive(:search_teaching_events_grouped_by_type) { events_by_type }
+
+    allow_any_instance_of(GetIntoTeachingApiClient::TeachingEventBuildingsApi).to \
+      receive(:get_teaching_event_buildings) { [] }
+  end
+
+  scenario "Submit a new form" do
+    allow_any_instance_of(GetIntoTeachingApiClient::TeachingEventsApi).to \
+      receive(:upsert_teaching_event) { [] }
+
+    navigate_to_new_submission
+
+    enter_valid_event_details
+
+    click_button "Submit for review"
+    expect(page).to have_text "Event submitted for review"
+  end
+
+  scenario "Edit a pending event with no building" do
+    allow_any_instance_of(GetIntoTeachingApiClient::TeachingEventsApi).to \
```

Comment:
It looks like these two allow calls are repeated, worth wrapping the scenarios in a context and putting them in a before block to dry it up?